### PR TITLE
Validate type name input (fixes #81)

### LINKED ISSUE #81: Does not deal well when creating Dexterity content types with two words
How to replicate this issue.

Create a new package using mr.bob, when asked select "Dexterity" then name the content type something like "Car Part".

After adding it to you buildout and relaunching Plone you'll get an error like this:

     SyntaxError: invalid syntax (interfaces.py, line 14)

You will see that interfaces.py contains a class with a space in it (ICar Part) in this case, which is not valid for Python.

### PR DESCRIPTION
**Pull request by pbauer**

### CODE REVIEW

File: bobtemplates/hooks.py [hunk 1]
```diff
@@ -141,6 +143,14 @@ def pre_dexterity_type_name(configurator, question):
         raise SkipQuestion
 
 
+def post_dexterity_type_name(configurator, question, answer):
```
Comment (tisto) [inline bobtemplates/hooks.py]: @pbauer wouldn't it make sense to write a few unit tests to show how this is supposed to work and to make sure the function works as expected?

**Comment by tisto** [comment]
@pbauer what's the status of this pr? Would you mind rebasing and adding a few tests?

**Comment by pbauer** [comment]
@tisto done

File: tests.py [hunk 2]
```diff
@@ -182,3 +183,27 @@ class HooksTest(unittest.TestCase):
     def test_to_boolean(self):
         # Initial simple test to show coverage in hooks.py.
         self.assertEqual(hooks.to_boolean(None, None, 'y'), True)
+
+    def test_post_dexterity_type_name(self):
+        """Test validation of extered dx type names
```
Comment (gforcada) [inline tests.py]: typo *extered* :-)

**Pull request merged**